Sources/race.swift

```diff
@@ -12,6 +13,16 @@
  - Warning: If any of the provided promises reject, the returned promise is rejected.
 */
 public func race<T>(promises: Promise<T>...) -> Promise<T> {
+    return try! race(promises)  // race only throws when the array param is empty, which is not
+                                // possible from this variadic paramater version
+}
+
+public func race<T>(promises: [Promise<T>]) throws -> Promise<T> {
+    guard !promises.isEmpty else {
+        let message = "Cannot race with an empty list of runners (Promises)"
+        throw NSError(domain: PMKErrorDomain, code: PMKInvalidUsageError, userInfo: ["messaage": message])
+    }
```

Comment:
IMO it's an error to race an empty array. `when` can logically work with an empty array since whatever the count it completes when they all complete, if there are no promises to complete, it is already completed. With `race` *one* must succeed so if you feed none, it will fail.